modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/AsyncResultSetImpl.java

```diff
@@ -132,12 +133,12 @@ public CompletableFuture<? extends AsyncResultSet<T>> fetchNextPage() {
         if (!hasMorePages()) {
             return (CompletableFuture<? extends AsyncResultSet<T>>) HAS_NO_MORE_PAGE_FUTURE;
         } else {
-            return cur.requestNextAsync(pageSize)
+            return IgniteExceptionMapperUtil.convertToPublicFuture(cur.requestNextAsync(pageSize)
```

Comment:
I see that "requestNextAsync" already wraps exception, do we really need it wrap again?

I think we need to remove the mapping from `requestNextAsync` or from here.

Reply:
oh... my bad, you are right. Thanks for your corrosiveness )